### PR TITLE
Adds config options to disable any borg module

### PR DESCRIPTION
**Pull request by Buggy123**
A̶l̶s̶o̶ ̶d̶i̶s̶a̶b̶l̶e̶s̶ ̶t̶h̶e̶ ̶e̶n̶g̶i̶b̶o̶r̶g̶ ̶b̶y̶ ̶d̶e̶f̶a̶u̶l̶t̶ ̶b̶e̶c̶a̶u̶s̶e̶ ̶i̶t̶ ̶c̶a̶n̶ ̶c̶r̶e̶a̶t̶e̶ ̶i̶n̶f̶i̶n̶i̶t̶e̶ ̶r̶e̶s̶o̶u̶r̶c̶e̶s̶

> maybe i'm biased but why not remove engiborgs then? :^)

### CODE REVIEW

**Comment by PKPenguin321** [comment]
>Also disables the engiborg by default because it can create infinite resources

the ol' bait and switch

**Comment by Supermichael777** [comment]
Standered is to leave something in the state it was before the change, else be bogged down in discussion. For example i think engie borg ads value to the game and so don't want you disabling it despite this being good.

**Comment by Buggy123** [comment]
Kay

**Comment by GunHog** [comment]
These config options seem rather pointless to me.

**Comment by PKPenguin321** [comment]
to you, sure

**Comment by bear1ake** [comment]
How to make cyborgs if standard module is forbidden?

**Comment by RemieRichards** [comment]
@bear1ake the default borg just looks standard, it just means you won't be able to /pick/ standard.

**Comment by bear1ake** [comment]
At this. +1

File: code/controllers/configuration.dm [hunk 1]
```diff
@@ -92,7 +92,15 @@
 	var/humans_need_surnames = 0
 	var/allow_random_events = 0			// enables random events mid-round when set to 1
 	var/allow_ai = 0					// allow ai job
+
+	var/forbid_standardborg = 0			// disallow standardborg module to be chosen.
```
Comment (Iamgoofball) [inline code/controllers/configuration.dm]: holy copypasta batman

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 2]
```diff
@@ -166,7 +166,19 @@
 	if(module)
 		return
 
-	var/list/modulelist = list("Standard", "Engineering", "Medical", "Miner", "Janitor","Service")
+	var/list/modulelist = list()
+	if(!config.forbid_standardborg)
```
Comment (Iamgoofball) [inline code/modules/mob/living/silicon/robot/robot.dm]: holy copypasta batman

File: code/modules/mob/living/silicon/robot/robot.dm [hunk 3]
```diff
@@ -166,7 +166,19 @@
 	if(module)
 		return
 
-	var/list/modulelist = list("Standard", "Engineering", "Medical", "Miner", "Janitor","Service")
+	var/list/modulelist = list()
+	if(!config.forbid_standardborg)
+		modulelist += "Standard"
```
Comment (Iamgoofball) [inline code/modules/mob/living/silicon/robot/robot.dm]: these shouldn't be strings

refactor borg code into datum modules

Comment (Buggy123) [inline code/modules/mob/living/silicon/robot/robot.dm]: I don't know how.

Comment (Iamgoofball) [inline code/modules/mob/living/silicon/robot/robot.dm]: ask in #coderbus they can help you

**Comment by KorPhaeron** [comment]
Ask me to reopen this if you ever figure it out/want to work on it again

**Pull request closed without merging**